src/reader/mod.rs

```diff
@@ -454,13 +457,14 @@ struct InterlaceIterator {
 impl iter::Iterator for InterlaceIterator {
     type Item = usize;
 
+    #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        if self.len == 0 || self.pass > 3 {
+        if self.len == 0 {
             return None
         }
-        let mut next = self.next + [8, 8, 4, 2][self.pass];
+        let mut next = self.next + *[8, 8, 4, 2].get(self.pass)?;
         while next >= self.len {
-            next = [4, 2, 1, 0][self.pass];
+            next = *[4, 2, 1, 0].get(self.pass)?;
```

Comment:
Please add a debug assert that `self.pass` is in bounds, and a comment explaining why the debug assert won't be hit (and why we did this optimization).